### DISCUSSION TITLE
Cannot delete DNN Deprecated Web Controls Library

### DISCUSSION THREAD

**Discussion by jmolhava** [dnnsoftware/Dnn.Platform#4298] [Q&A] [answered]
I just upgraded to DNN 9.8.0 and followed the steps to remove Telerik. Everything went well except for one step:
9) In Extensions > Libraries, uninstall DNN Deprecated Web Controls Library (Tick delete files).

I ran the the SQL query and even checked the database so it is definitely NOT a system package, however there is no option to delete it. 

Can someone please tell me how to delete this package?

**Comment by moorecreative** [comment]
@valadas this sounds like the same steps that I ran through in testing with you the night of the SoFri meeting... 

@jmolhava there are notes in this thread:
https://github.com/dnnsoftware/Dnn.Platform/issues/4199

**Reply by jmolhava** [reply]
Thank you! I followed the step in that thread beginning with GetPackageDependencies, and then manually removed the dependency from the database. Then I was able to uninstall the library.

**Comment by valadas** [comment, marked as answer]
And, the script in the release notes has been updated to cover for this issue too, the new script should handle this now.

**Comment by thienvc** [comment]


**Reply by mitchelsellers** [reply]
PLEASE do not post security related items in this medium and follow our published guidelines for security-related information disclosure.

**Reply by thienvc** [reply]
> PLEASE do not post security related items in this medium and follow our published guidelines for security-related information disclosure.

But such serious (not new) security flaws, but seem to be underestimated, should alert people to the urgent need to upgrade and eliminate telerik control entirely.

**Reply by mitchelsellers** [reply]
Alerting people to upgrade is one thing, posting a direct link to how sites can be hacked in another.

We have a security policy, I would respectfully request that it be followed 

I don't know how much more apparent we can make it that upgrading is important

**Reply by valadas** [reply]
Exactly, we do advice everyone to keep their DNN updated, as for any other software this is a must for keeping up with security fixes. We also have the DNN Security Center for awareness, but we want to avoid publishing DETAILS on how to hack something, this is not technically wrong but it needs to happen on private channels, that is why we have an email dedicated to security issues where we can share details or new issues without exposing those details to the public and all potential hackers.
